### DISCUSSION TITLE
TS2307: Cannot find module '@mantine/carousel' or its corresponding type declarations.

### DISCUSSION THREAD

**Discussion by VermilionB** [mantinedev/mantine#5310] [Q&A] [answered]
When I try to use Carousel component it giving me an error:
TS2307: Cannot find module '@mantine/carousel' or its corresponding type declarations.
I used code from docs, how can I solve this problem?

**Comment by rtivital** [comment]
You need to install `@mantine/carousel` package. See https://mantine.dev/others/carousel/#installation

**Reply by VermilionB** [reply]
I have it installed, it specifically complains about type declaration
![image](https://github.com/mantinedev/mantine/assets/73846126/b6278199-a74c-445a-9505-aa3523400af3)

**Reply by rtivital** [reply, marked as answer]
It works correctly on codesandbox with TypeScript – https://codesandbox.io/s/hungry-tess-jtv5sz?file=/src/App.tsx
Make sure that your package manager actually installed the dependency and your editor/bundler has access to it.

**Reply by VermilionB** [reply]
I added some dependencies from this codesandbox and editted my tsconfig.json and it works for me, thanks

**Comment by Utpal7561** [comment]
If my data.json contain more then 6 items, it show problem in slideSize  
 <Carousel
            slideSize={ { base: '100%', sm: '50%', md: '33.333333%' }}
            slideGap={{ base: 'xl', sm: 'md'}}
            align="start"
            slidesToScroll={mobile ? 1: 1}
            loop
            >
please look that issue
